src/components/hoc/withLoanAccount.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react'
+
+import { LoanAccount } from '../../controllers/loan-manager/types'
+import { useSelector } from '../../types/reactRedux'
+import { LoadingScene } from '../scenes/LoadingScene'
+
+interface NavigationProps {
+  navigation: { popToTop: () => void }
+  route: { params: { loanAccountId: string } }
+}
+
+type WithoutLoanAccount<Props> = Omit<Props, 'loanAccount'>
+
+/**
+ * Looks up a loanAccount for a scene.
+ * If the loanAccount is missing, replaces the scene with a spinner instead.
+ */
+export function withLoanAccount<Props extends { loanAccount: LoanAccount }>(
+  Component: React.ComponentType<Props>
+): React.FunctionComponent<WithoutLoanAccount<Props> & NavigationProps> {
+  return (props: WithoutLoanAccount<Props> & NavigationProps) => {
+    const { navigation, route } = props
+
+    const loanAccounts = useSelector(state => state.loanManager.loanAccounts)
+    const loanAccount = loanAccounts[route.params.loanAccountId]
+
+    React.useEffect(() => {
+      if (loanAccount == null) navigation.popToTop()
+    }, [navigation, loanAccount])
+
+    if (loanAccount == null) return <LoadingScene />
+    return <Component {...(props as any)} loanAccount={loanAccount} />
```

Comment:
Redirect only when 'loanAccount == null' && 'loanAccounts != null'
Loading state when 'loanAccounts == null'

Reply:
I realized that `loanAccounts === null` is never a case. Instead, we should introduce some state to indicate that the loanAccounts are still initializing or loading. I think that's a break away task though, and luckily not necessary for this all to work until we have deep linking to specific loan accounts (maybe when we improve push notification routing).